services/characterlcdscreen.md

```diff
@@ -0,0 +1,49 @@
+# Character LCD Screen
+
+    identifier: 0x1f37c56a
+
+A character LCD screen.
+
+## Registers
+
+    rw message: string @ value
+
+Sets text, with `\n` as the newline character.
+
+    rw cursor: bool @ 0x80
+
+Configures to show or hide the cursor.
+
+    rw blink: bool @ 0x81
+
+Configures to blink the cursor.
+
+    enum TextDirection : u8 {
+        LeftToRight = 1,
+        RightToLeft = 2
+    }
+    rw text_direction: TextDirection @ 0x82
+
+Specifies the RTL or LTR direction of the text.
+
+    rw backlight_color? @ 0x83 {
+        red: u8
+        green: u8
+        blue: u8
+    }
+
+If available, sets the backlight color.
+
+    const rows?: u8 @ 0x180
+
+Gets the number of rows. Default is ``2``.
```

Comment:
I don't think this should be optional - I don't see why one would skip it. 

Also, you can use syntax `rows = 2: u8` to specify value at reset, but I guess we should now do that for const registers (since they never change)